smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/CodegenVisitor.kt

```diff
@@ -88,6 +89,7 @@ class CodegenVisitor(context: PluginContext) : ShapeVisitor.Default<Void>() {
         resolvedModel = AddOperationShapes.execute(resolvedModel, settings.getService(resolvedModel), settings.moduleName)
         resolvedModel = RecursiveShapeBoxer.transform(resolvedModel)
         resolvedModel = NestedShapeTransformer.transform(resolvedModel, settings.getService(resolvedModel))
+        resolvedModel = FlattenServiceErrorsOntoOperations.transform(resolvedModel, settings.getService(resolvedModel))
```

Comment:
While this approach would work, and even has a certain elegance about it, my concern is with how this will scale:
- For a number of errors `m` and a number of operations `n`, this will grow the generated codebase at a rate of `m * n`.  For a service with many operations (which we have in spades) and with many errors modeled at the service level (which we don't right now but I could easily see happening in the future) this will expand the generated codebase excessively.

An alternate approach:
- Code-generate a static method for each service that attempts to turn a response into one of the errors defined on the service.  The method should either return an error, or `nil` if no error modeled on the service can be created from the response.
- When parsing an error from each operation on the service, call that static method.  If an error is returned, throw it; else continue parsing the error the same way we currently do.

The advantage of this approach is that it would grow the codebase only proportionally to the number of service errors.

Reply:
Decided to go with alternate approach as suggested here after offline discussion.